### PR TITLE
Adds a more descriptive error msg for wrong wrapping

### LINKED ISSUE #4498: More descriptive error msg for certificate loading
When we fail to load a PEM certificate we do this: https://github.com/pyca/cryptography/blob/master/src/cryptography/hazmat/backends/openssl/backend.py#L1151

Given that people occasionally attempt to load valid certs that don't have proper line wrapping (see: https://github.com/pyca/cryptography/issues/4487 and https://github.com/pyca/cryptography/issues/4085) should we alter the text to say something about how what a valid PEM looks like and that they should check their cert?

### PR DESCRIPTION
**Pull request by andrealmeid**
I made a proof of concept of a function that should fix #4498. I addressed my code to check specifically for a bad wrapped RSA 512 certificate, verifying positions to ensure that it has a `\n` there. I have some questions about my code:

- This function is OK like that or should it be more generic than this?
- Is my logic right? 
- Where should I place it? I think the code would be cleaner if it has a `pem_check_wrap()`  function.
- There are performance issue in this code? I could remove the `chr()` cast for example. I inserted it to make the logic more obvious. 

I also want to write a unit test in the future to match this  specific case. Thanks!

### CODE REVIEW

**Comment by reaperhulk** [comment]
Hi @andrealmeid, thanks for taking this task on. I think this approach is more complex than we want though -- rather than try to determine why the key isn't loading via checking what byte values are at specific offsets we should just change the ValueError text to something like "Unable to load certificate, is it a valid PEM?". We could add more there (like that PEM files have BEGIN and END lines and no more than 64 base64 encoded characters per line between them) or we could potentially write up a longer explanation of what a PEM is in our docs and then put a link to that in the error msg.

**Comment by andrealmeid** [comment]
Ok @reaperhulk! I think will take the docs approach, filled with some examples. May I change `ValueError` messages for every function that imports a PEM file to link that docs page? This would address issues involving PEM in certificates, certificates requests, keys, ...

**Comment by reaperhulk** [comment]
If you want to do the docs approach that's fine. It might be best to add it as a FAQ entry.

**Comment by alex** [comment]
Looks like there's a flake8 issue.

**Comment by andrealmeid** [comment]
Thanks @alex, should be fixed now.

File: docs/faq.rst [hunk 1]
```diff
@@ -111,6 +111,37 @@ not yet possible you  can also install ``cryptography`` with
...
+
+PEM is a format (defined by several RFCs, but originally :rfc:`1421`) for
+encoding keys, certificates and others cryptographic data into a regular form.
+The data is encoded as base64 and wrapped with a header and footer.
+
+If you are having trouble importing PEM files, make sure your file fits
+the following rules:
+
+* has a one-line header like this: ``-----BEGIN [FILE TYPE]-----``
+  (where ``[FILE TYPE]`` is certificate, public key, et cetera)
```
Comment (alex) [inline docs/faq.rst]: Please use real examples here, ``CERTIFICATE``, ``PUBLIC KEY``.

**Pull request merged**